### PR TITLE
#1161: fix `no implicit conversion of Symbol into Integer` in `quantity-of-deliverables` judge

### LINKED ISSUE #1161: Implicit conversion of Symbol into Integer fails in total_issues_created.rb
```
TypeError: no implicit conversion of Symbol into Integer
   	/action/judges/quantity-of-deliverables/total_issues_created.rb:21:in 'block (2 levels) in Object#total_issues_created'
   	/usr/local/bundle/gems/sawyer-0.9.2/lib/sawyer/resource.rb:128:in 'Hash#each'
   	/usr/local/bundle/gems/sawyer-0.9.2/lib/sawyer/resource.rb:128:in 'Sawyer::Resource#each'
   	/action/judges/quantity-of-deliverables/total_issues_created.rb:19:in 'block in Object#total_issues_created'
```

### PR DESCRIPTION
**Pull request by Yegorov**
Closes #1161 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved resilience when processing issue data from GitHub API by correctly handling edge cases with malformed issue items.

* **Tests**
  * Added test coverage to verify the system properly processes issues even when data format is unexpected.

### CODE REVIEW

File: test/judges/test-quantity-of-deliverables.rb [hunk 1]
```diff
@@ -357,4 +357,50 @@ def test_quantity_of_deliverables_fix_gap
...
+      'https://api.github.com/repos/foo/foo/commits?per_page=100&since=2024-07-15T21:00:00%2B00:00',
+      body: []
+    )
+    stub_github(
+      'https://api.github.com/repos/foo/foo/issues?per_page=100&since=%3E2024-07-15',
+      body: [
+        {
+          pull_request: {}
+        },
+        []
```
[the remark this answers is not in the document]
Comment (Yegorov) [reply test/judges/test-quantity-of-deliverables.rb]: @yegor256 Yes, it looks really strange, but this response corresponds to the exception in the issue.

> how could this happen?

This is most likely a bug in Github API. If an array element is an array, [it will remain an array](https://github.com/lostisland/sawyer/blob/7a60719e675f79d5569ff3e4729d58c03fb64831/lib/sawyer/response.rb#L41).

> GitHub API can return this?

Of course [not](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#list-repository-issues)

What surprises me most is that we didn't receive an exception for the 500 status code (or any other). That is, it was a valid response with a 200 code.

**Comment by yegor256** [comment]
This pull request has merge conflicts. Please rebase it on the default branch to resolve them so it can be merged.

**Comment by yegor256** [comment]
We are closing this pull request, because it has merge conflicts with `master` and has been open for more than a month. Please, forgive us for this: it says nothing about the quality of your work, it's just housekeeping.

If the change still matters to you, please submit a new pull request. It may be very similar to this one, but it has to take into account everything that landed in `master` since then.

Thank you for contributing, and sorry again for the inconvenience!
